src/Community/triangle_count.jl

```diff
@@ -0,0 +1,80 @@
+"""
+    triangle_count(g)
+
+Return total number of triangles in the undirected graph `g`.
+
+### References
+- [One Quadrillion Triangles Queried on One Million Processors](https://ieeexplore.ieee.org/document/8916243).
+"""
+@traitfn function triangle_count(g::AG::(!IsDirected)) where {T, AG<:AbstractGraph{T}}
+    ntri = 0
+    # create a degree-ordered directed graph where the original
+    # undirected edges are directed from low-degree to high-degree
+    adjlist = [T[] for _ in vertices(g)]
+    for u in vertices(g)
+        for v in neighbors(g, u)
+            degv = degree(g, v)
+            degu = degree(g, u)
+            # add edge u => v in pruned graph only if degv > degu
+            # (or v > u for tie breaking)
+            if degv > degu || (degv == degu && v > u)
+                push!(adjlist[u], v)
+            end
+        end
+    end
+    for u in vertices(g)
+        adju = adjlist[u]
+        lenu = length(adju)
+        # chose u as pivot and check all pairs of its neighbors
+        for i = 1:lenu, j = i+1:lenu
+            v = adju[i]
+            w = adju[j]
+            # for a pair of edges u => v & u => w, if there is an edge
+            # v => w in the original graph then we've found a triangle
+            if has_edge(g, v, w)
```

Comment:
`has_edge` does a check to see whether v > w. We might be able to eliminate this check. This could be generalized by modifying `has_edge` to call an (inlined) version of `unordered_has_edge` that we can create.

Reply:
I don't think it does https://github.com/JuliaGraphs/LightGraphs.jl/blob/743ec560f28a22c38bebee33d0cdf2fc78f302bb/src/SimpleGraphsCore/simplegraph.jl#L395

Reply:
It does the check here: https://github.com/JuliaGraphs/LightGraphs.jl/blob/743ec560f28a22c38bebee33d0cdf2fc78f302bb/src/SimpleGraphsCore/simplegraph.jl#L400